docs/src/proposals/unified-scheduler.md

```diff
@@ -0,0 +1,96 @@
+# Unified Scheduler
+
+## Architechtural problem
+
+- there's two execution scheduling algorithms, and both are subotimmal for current tx load. 
+independent algorithms are exposing the so-called replayability risk
+
+also, increasing banking threads doesn't linearly scale to the number of cpu cores.
+ 
+## Current Transaction load pattern
+
+## Redefined scheduler's design goal
+
+saturate N cores with full of tx exec cycles.
+
+## Proposed Solution
+
+## why unify?: because simply using the same algo makes most sense.
+
+## Currnet implementation problem
+
+- why batching isn't good?
+- what about block max compute unit?
+
+## Proposed implementation
+
+### high-level scheduling algorithm
+
+eventual successful locking.
+
+pessimic locking is chosen for now (might switch to optimistic)
+
+
+### design philophy
+
+- no timer/no bucketing (thus, no jitter and no odd juggy incentive)
+- no huetrics (try to be generic/agnostic/adaptive as much as possible)
+- no estimator/prediction (so bots can't trick leaders)
+- single threaded
+- determinicity
+- strict fairness, only considering priority_fee/fcfs
+- approx. O(n) where n is gross total of addresses in transactions.
+- strict adherance to local fee market
+- censorship resistent
+- 100k scheduling/s
+- highly contended address with 1m pending txes doesn't affect overall performance at all.
```

Comment:
I've talked with @taozhu-chicago about this, likely we'll want to move the QoS checks for account limits up into the scheduler:
1. scheduler will already be looking at accounts locked by the tx, so easy to fit in per-account so we could track account costs as we schedule
2. this could allow us to drop lower-priority tx early in the scheduling stage so we don't even add them into the scheduling structures

good goal to have, but if we inlcude these changes we probably won't have to deal with 1m pending txs on the same address

Reply:
ah, but this would conflict with your goal of "no estimator/prediction" 😅